identity/keystore_filesystem.go

```diff
@@ -34,26 +35,46 @@ import (
 	"golang.org/x/crypto/hkdf"
 )
 
+type ks interface {
+	Accounts() []accounts.Account
+	NewAccount(passphrase string) (accounts.Account, error)
+	Find(a accounts.Account) (accounts.Account, error)
+	Unlock(a accounts.Account, passphrase string) error
+	Lock(addr common.Address) error
+	SignHash(a accounts.Account, hash []byte) ([]byte, error)
+	Export(a accounts.Account, passphrase, newPassphrase string) (keyJSON []byte, err error)
+}
+
 // NewKeystoreFilesystem create new keystore, which keeps keys in filesystem.
 func NewKeystoreFilesystem(directory string, lightweight bool) *Keystore {
-	var ks *ethKs.KeyStore
-	if lightweight {
+	var ks ks
+	var keyDecryptFunc func(keyjson []byte, auth string) (*ethKs.Key, error)
+
+	// In test scenarios, we'll skip the real ethereum keystore and prefer a memory one instead.
+	if flag.Lookup("test.v") != nil {
```

Comment:
This e2e test is failing https://github.com/mysteriumnetwork/node/blob/master/e2e/mobile_entrypoint_test.go as it needs real keystore. 

Maybe it's better to create separate func `NewKeystoreMock() *Keystore` and use it explicitly there needed.